pkg/pillar/diskmetrics/usage.go

```diff
@@ -201,3 +203,48 @@ func PersistUsageStat(log *base.LogObject) (*types.UsageStat, error) {
 	}
 	return usageStat, nil
 }
+
+// FoundLargeFiles is returned by FindLargeFiles
+type FoundLargeFiles struct {
+	Path string
+	Size int64
+}
+
+// FindLargeFiles walks a directory and reportes all large files unless
+// they are in an excluded directory
+func FindLargeFiles(root string, limit int64, excludePaths []string) ([]FoundLargeFiles, error) {
+	var list []FoundLargeFiles
+	walkErr := filepath.WalkDir(filepath.Clean(root), func(path string, di fs.DirEntry, err error) error {
+		// We skip all directories
+		if di.IsDir() {
+			return nil
```

Comment:
I think it makes sense to check here if we are entering a directory that should be skipped, and not go any further if that is the case. In the current implementation, as I understand it, we always go through all directories and check all files. And only when checking a file do we decide not to count it. So if we have a directory to skip with a huge amount of files in it - we will still check them all.
It's not critical for us, as it's not a performance-demanding piece of code, but it still looks a bit strange...

Reply:
Fixed.